### PR TITLE
ambient: rename default waypoint (again)

### PR DESCRIPTION
**Pull request by howardjohn**
Before: `default-namespace`, which ends up suffixing to `default-namespace-istio-waypoint`

Now: `waypoint`, with suffix disabled. So its just `waypoint`.

From discussion with @louiscryan

### CODE REVIEW

**Review by louiscryan — APPROVED** [review]
LGTM. Much more sensible

**Review by hzxuzhonghu — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Review by stevenctl — APPROVED** [review]

File: pilot/pkg/config/kube/gateway/testdata/deployment/waypoint-no-network-label.yaml [hunk 1]
```diff
@@ -188,7 +188,7 @@ spec:
           name: istio-token
         - mountPath: /etc/istio/pod
           name: istio-podinfo
-      serviceAccountName: namespace-istio-waypoint
+      serviceAccountName: namespace
```
Comment (costinm) [inline pilot/pkg/config/kube/gateway/testdata/deployment/waypoint-no-network-label.yaml:191]: The service account is named 'namespace' ???

**Review by costinm — COMMENTED** [review]
Better than before - but not sure if I understand all implications, in particular since it seems we disable the suffix we use for gateways.

Does it mean we can't have more than one, and canary will not be possible ? Is the name used anywhere in the code or by tools - or purely a naming convention ? 

Would it be possible to make it configurable - and maybe have a test with a different name so it don't become another magic API ?

**Comment by howardjohn** [comment]
@costinm its not a fixed name. Its the mapping from the `Gateway` name to the generated resources (svc, deployment, sa).

if the Gateway is named `waypoint` everything else is `waypoint`. If its `namespace` (like the test case), its `namespace (bad name, but its a test -- not what we tell users to do).

**Comment by costinm** [comment]
Unless they call it 'default'...

If I remember correctly, the suffix was added to allow multiple revisions of Istiod  acting as controllers ? Or something along the line - that's what I'm trying to understand - remember all the pain when istiod-revision was added.

Do we expect waypoint will only have one controller - and not use revisions ? And even with one controller - having a 'canary' would improve reliability.

**Pull request merged**